website/docs/d/vm_sizing_policy.html.markdown

```diff
@@ -0,0 +1,35 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vm_sizing_policy"
+sidebar_current: "docs-vcd-datasource-vm-sizing-policy"
+description: |-
+  Provides a vCloud Director VM sizing policy data source. This can be
+  used to read VM sizing policy.
+---
+
+# vcd\_vm\_sizing\_policy
+
+Provides a vCloud Director VM sizing policy data source. This can be
+used to read VM sizing policy.
+
+Supported in provider *v3.0+*
+
+## Example Usage
+
+```hcl
+data "vcd_vm_sizing_policy" "tf-policy-name" {
+  name = "my-rule"
+}
+output "policyId" {
+	value = data.vcd_vm_sizing_policy.tf-policy-name.id
+}
+```
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations
```

Comment:
I am unable to create a policy **when VDC does not exist yet**.

My config has many resources and most of them are in the "demo-org" which is created by the same config:

```
provider "vcd" {
  url      = "https://${var.vcd_host}/api"
  user     = "administrator"
  password = "${var.admin_pass}"
  sysorg   = "System"

  org = "demo-org"
  vdc = "demo-vdc"

  max_retry_timeout    = "120"
  allow_unverified_ssl = "true"

  logging = "true"
}
```

I understand that sizing policy doesn't belong to any specific Org or VDC, so I define it as such:

```
resource "vcd_vm_sizing_policy" "min_size" {
  org = "System" # <------ IMPORTANT

  name        = "min-size"
  description = "smallest size"

  cpu {
    limit_in_mhz          = "2400"
    count                 = "1"
    speed_in_mhz          = "2500"
    cores_per_socket      = "1"
    reservation_guarantee = "0.55"
    shares                = "886"
  }

  memory {
    size_in_mb            = "512"
    limit_in_mb           = "640"
    reservation_guarantee = "0.3"
    shares                = "886"
  }
}
```

However, `apply` fails:

```
$ terraform apply
data.vcd_org_user.org_admin: Refreshing state...
vcd_org.demo_org: Refreshing state... [id=urn:vcloud:org:33073d16-4bc5-4f2d-a86b-66555deafe5f]
vcd_external_network.demo_ext_net: Refreshing state... [id=urn:vcloud:network:db544616-a681-4b21-b1d8-614640e7550a]
vcd_vm_sizing_policy.min_size: Refreshing state... [id=urn:vcloud:vdcComputePolicy:6e06d3cf-6594-4a44-a711-b216b8e51862]

Error: error retrieving Org: error retrieving VDC demo-vdc: [ENF] entity not found
```

Looks like `vcd_vm_sizing_policy` resource is using specific VDC from `provider.vdc` value, even though it shouldn't care about VDC.

Sharing more widely as this looks quite an interesting case overall.
CC: @Didainius @dataclouder

Reply:
fixed

Reply:
It works! Thank you.